scripts/hud/synchronizer.js

```diff
@@ -0,0 +1,349 @@
+					this.getBufferedSum(this.syncDeltaHistory) *
+					(this.mom_hud_synchro_flip_enable == 1 ? -1 : 1);
+				this.firstPanelWidth = this.wrapValueToRange(this.firstPanelWidth, 0, this.maxSegmentWidth, true);
+
+				this.segments[0].style.width =
+					(isNaN(this.firstPanelWidth) ? this.maxSegmentWidth : this.firstPanelWidth.toFixed(3)) + '%';
+				this.segments.forEach((segment, i) => {
+					if (i > 0) this.segments[i].style.width = this.maxSegmentWidth + '%';
+					let index = i + (this.bIsFirstPanelColored ? 1 : 0);
+					segment.style.backgroundColor = index % 2 ? color : altColor;
+				});
+
+				// hide needle
+				this.needle.style.visibility = 'collapse';
+				break;
+		}
+	}
+
+	static getColorTuple(ratio, bOverStrafing) {
+		// cases where gain effectiveness is >90%
+		if (ratio > 1.02) return COLORS.EXTRA;
+		else if (ratio > 0.98) return COLORS.PERFECT;
+		else if (ratio > 0.95) return COLORS.GOOD;
+		else if (ratio <= -5.0) return COLORS.STOP;
+
+		const lerpColorTuples = (c1, c2, alpha) => {
+			return [
+				this.lerpColorStrings(c1[0], c2[0], alpha.toFixed(3)),
+				this.lerpColorStrings(c1[1], c2[1], alpha.toFixed(3))
+			];
+		};
+
+		// cases where gain effectiveness is <90%
+		if (!bOverStrafing) {
+			if (ratio > 0.85) return lerpColorTuples(COLORS.SLOW, COLORS.GOOD, (ratio - 0.85) / 0.1);
+			else if (ratio > 0.75) return COLORS.SLOW;
+			else if (ratio > 0.5) return lerpColorTuples(COLORS.NEUTRAL, COLORS.SLOW, (ratio - 0.5) / 0.25);
+			else if (ratio > 0.0) return COLORS.NEUTRAL;
+			else if (ratio > -5.0) return lerpColorTuples(COLORS.NEUTRAL, COLORS.STOP, Math.abs(ratio) / 5.0);
+		} else {
+			if (ratio > 0.8) return lerpColorTuples(COLORS.SLOW, COLORS.GOOD, (ratio - 0.8) / 0.15);
+			else if (ratio > 0.0) return lerpColorTuples(COLORS.LOSS, COLORS.SLOW, (ratio - 0.25) / 0.55);
+			else if (ratio > -5.0) return lerpColorTuples(COLORS.LOSS, COLORS.STOP, Math.abs(ratio) / 5.0);
+		}
+	}
+
+	static wrapValueToRange(value, min, max, bShouldTrackWrap) {
+		const range = max - min;
+		while (value > max) {
+			value -= range;
+			if (bShouldTrackWrap) {
+				this.bIsFirstPanelColored = !this.bIsFirstPanelColored; // less than clean way to track color flips
+			}
+		}
+		while (value < min) {
+			value += range;
+			if (bShouldTrackWrap) {
+				this.bIsFirstPanelColored = !this.bIsFirstPanelColored;
+			}
+		}
+		return value;
+	}
+
+	static findFastAngle(speed, maxSpeed, maxAccel) {
+		const threshold = maxSpeed - maxAccel;
+		return Math.acos(speed < threshold ? 1 : threshold / speed);
+	}
+
+	static getSize(vec) {
+		return Math.sqrt(this.getSizeSquared(vec));
+	}
+
+	static getSizeSquared(vec) {
+		return vec.x * vec.x + vec.y * vec.y;
+	}
+
+	static initializeBuffer(size) {
+		let buffer = new Array(size);
+		buffer.fill(0);
+		return buffer;
+	}
+
+	static addToBuffer(buffer, value) {
+		buffer.push(value);
+		buffer.shift();
+	}
+
+	static getBufferedSum(history) {
+		return history.reduce(function (sum, element) {
```

Comment:
I'd prefer we stay consistent here and always use arrow functions for nested functions like this, so 
```js
(sum, element) => {
```
rather than
```js
function (sum, element) {
```